common/persistence/sql/sqlExecutionManager.go

```diff
@@ -95,7 +96,14 @@ func (m *sqlExecutionManager) createWorkflowExecutionTx(tx sqldb.Tx, request *p.
 
 	lastWriteVersion := common.EmptyVersion
 
-	if row != nil {
+	// current workflow record check
+	if row == nil {
```

Comment:
Should this proceed if the row is nil and the create workflow mode is not zombie